presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -145,6 +148,17 @@ protected PlanBuilder visitQuerySpecification(QuerySpecification node, Void cont
         return builder;
     }
 
+    private Set<SubqueryExpression> getSubqueryExpressionsNotUsedInInPredicates(QuerySpecification node)
```

Comment:
maybe move this method to `Analysis`. And replace the method `getScalarSubqueries`.